core/common/src/main/java/alluxio/ConfigurationDocGenerator.java

```diff
@@ -0,0 +1,135 @@
+   *
+   * @param defaultKeys Collection which is from PropertyKey DEFAULT_KEYS_MAP.values()
+   * @param filePath    path for csv files
+   */
+  public static void writeCSVFile(Collection<? extends PropertyKey> defaultKeys, String filePath)
+      throws IOException {
+    if (defaultKeys.size() == 0) {
+      return;
+    }
+
+    //CSV file header
+    String fileHeader = "propertyName,defaultValue";
+    FileWriter fileWriter;
+
+    //HashMap for CSV file names
+    Map<String, String> fileNames = new HashMap<>();
+    fileNames.put("user", PathUtils.concatPath(filePath, "user-configuration.csv"));
+    fileNames.put("master", PathUtils.concatPath(filePath, "master-configuration.csv"));
+    fileNames.put("worker", PathUtils.concatPath(filePath, "worker-configuration.csv"));
+    fileNames.put("security", PathUtils.concatPath(filePath, "security-configuration.csv"));
+    fileNames.put("keyvalue", PathUtils.concatPath(filePath, "key-value-configuration.csv"));
+    fileNames.put("common", PathUtils.concatPath(filePath, "common-configuration.csv"));
+
+    //HashMap for FileWriter per each category
+    Map<String, FileWriter> fileWriterMap = new HashMap<>();
+    Closer closer = Closer.create();
+    try {
+      for (Map.Entry<String, String> entry : fileNames.entrySet()) {
+        fileWriter = new FileWriter(entry.getValue());
+        //Write the CSV file header
+        fileWriter.append(fileHeader);
+        //Add a new line separator after the header
+        fileWriter.append("\n");
+        //put fileWriter
+        fileWriterMap.put(entry.getKey(), fileWriter);
+        //register file writer
+        closer.register(fileWriter);
+      }
+
+      for (PropertyKey iteratorPK : defaultKeys) {
+        String pKey = iteratorPK.toString();
+        String value;
+        PropertyKey pk = new PropertyKey(pKey);
+        if (pk.getDefaultValue() == null) {
+          value = "";
+        } else {
+          value = pk.getDefaultValue();
+        }
+
+        //Persist property key and default value to CSV
+        String keyValueStr = pKey + "," + value + "\n";
+        if (pKey.contains(".user.")) {
+          fileWriter = fileWriterMap.get("user");
+        } else if (pKey.contains(".master.")) {
+          fileWriter = fileWriterMap.get("master");
+        } else if (pKey.contains(".worker.")) {
+          fileWriter = fileWriterMap.get("worker");
+        } else if (pKey.contains(".security.")) {
+          fileWriter = fileWriterMap.get("security");
+        } else if (pKey.contains(".keyvalue.")) {
+          fileWriter = fileWriterMap.get("keyvalue");
+        } else {
+          fileWriter = fileWriterMap.get("common");
+        }
+        fileWriter.append(keyValueStr);
+      }
+
+      LOG.info("Property Key CSV files were created successfully.");
+    } catch (Exception e) {
+      throw closer.rethrow(e);
+    } finally {
+      try {
+        for (Map.Entry<String, FileWriter> entry : fileWriterMap.entrySet()) {
```

Comment:
you don't need line 112-114. `closer.close()` is sufficient as it will close all registered `Closable`s